src/Bicep.LangServer/Handlers/ImportKubernetesManifestHandler.cs

```diff
@@ -0,0 +1,213 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+using Bicep.Core.Diagnostics;
+using Bicep.Core.Parsing;
+using Bicep.Core.PrettyPrint;
+using Bicep.Core.PrettyPrint.Options;
+using Bicep.Core.Syntax;
+using Bicep.LanguageServer.Telemetry;
+using MediatR;
+using OmniSharp.Extensions.JsonRpc;
+using OmniSharp.Extensions.LanguageServer.Protocol.Server;
+using SharpYaml.Serialization;
+
+namespace Bicep.LanguageServer.Handlers
+{
+    [Method("bicep/importKubernetesManifest", Direction.ClientToServer)]
+    public record ImportKubernetesManifestRequest(string ManifestFilePath)
+        : IRequest<ImportKubernetesManifestResponse>;
+
+    public record ImportKubernetesManifestResponse(string? BicepFilePath);
+
+    public class ImportKubernetesManifestHandler : IJsonRpcRequestHandler<ImportKubernetesManifestRequest, ImportKubernetesManifestResponse>
+    {
+        private readonly TelemetryAndErrorHandlingHelper<ImportKubernetesManifestResponse> helper;
+
+        public ImportKubernetesManifestHandler(ILanguageServerFacade server, ITelemetryProvider telemetryProvider)
+        {
+            this.helper = new TelemetryAndErrorHandlingHelper<ImportKubernetesManifestResponse>(server.Window, telemetryProvider, new(null));
+        }
+
+        public Task<ImportKubernetesManifestResponse> Handle(ImportKubernetesManifestRequest request, CancellationToken cancellationToken)
+            => helper.ExecuteWithTelemetryAndErrorHandling(async () => {
+                var bicepFilePath = Path.ChangeExtension(request.ManifestFilePath, ".bicep");
+                var manifestContents = await File.ReadAllTextAsync(request.ManifestFilePath);
```

Comment:
Is this the right thing to do from the VS Code POV? I expected this to take a URI and need to interact with VS Code APIs to get the document. eg: will this work in remote scenarios?

Reply:
It's a shortcut - the UI side brings up a file picker, meaning that you can only operate on on-disk files. Feels good enough for now, as there's no way I'm aware of to access the 'unsafe' behavior.

If we wanted to extend this in future, we'd need to either have the extension be aware of .yml files, or have the LSP send and receive the full file contents instead of sending/receiving URIs.